github/resource_github_repository_custom_properties.go

```diff
@@ -67,6 +84,20 @@ func resourceGithubRepositoryCustomPropertiesCreateOrUpdate(d *schema.ResourceDa
 	repoName := d.Get("repository_name").(string)
 	properties := d.Get("property").(*schema.Set).List()
 
+	// Get all organization custom property definitions to determine types
+	orgProperties, _, err := client.Organizations.GetAllCustomProperties(ctx, owner)
```

Comment:
We need to fetch the organization level properties and check the type, since GitHub API accepts different values depending on type.

For single_select, true_false and string, it's a simple string.

For multi_select, it is an array.